src/engines/kotorbase/gui/dialog.cpp

```diff
@@ -56,6 +56,14 @@ DialogGUI::DialogGUI(Module &module) :
 void DialogGUI::startConversation(const Common::UString &name, Aurora::NWScript::Object *owner) {
 	try {
 		_dlg.reset(new Aurora::DLGFile(name, owner));
+		_dlg->setParameterCallback(
+				std::bind(
+						&DialogGUI::callbackParameter,
+						this,
+						std::placeholders::_1,
+						std::placeholders::_2
+				)
+		);
```

Comment:
Hmmm.. I'm not so sure about that.

Shouldn't the parameters belong to the script, and as such be a property of `Aurora::NCSFile` that can be set and queried?

That would get rid of the need to add a callback there. And I also kind want to add any external state to a script that can be overwritten by another script. (In the future, I think we will need to run several scripts somewhat in parallel, at least for the heartbeat scripts?)